src/main/java/universe/UniverseInferenceAnnotatedTypeFactory.java

```diff
@@ -30,9 +25,17 @@
 import checkers.inference.model.ConstraintManager;
 import checkers.inference.model.Slot;
 import checkers.inference.util.InferenceViewpointAdapter;
+import universe.qual.*;
 
 public class UniverseInferenceAnnotatedTypeFactory extends InferenceAnnotatedTypeFactory {
 
+    public final AnnotationMirror ANY = AnnotationBuilder.fromClass(elements, Any.class);
```

Comment:
Instead of duplicating all these fields, can't you access them through the `realTypeFactory`? That is, have a field of type `UniverseAnnotatedTypeFactory`, assign the `realTypeFactory` to that, and then access the qualifiers through that.

Reply:
Okay.